drivers: espi: xec: Fix unintended eSPI frequency override whenever default IO mode is changed

albertofloyd opened the pull request:
Default eSPI frequency 20MHz is used whenever Quad IO mode is advertised.
This due to incorrect register update whenever IO mode is different than default.

Update sample code to use valid non-default eSPI frequency such as 25MHz

![image](https://user-images.githubusercontent.com/16172078/90938052-ff9c1f80-e3bc-11ea-90e7-adb09cfb8c72.png)

albertofloyd committed:
drivers: espi: xec: Fix eSPI frequency override during IO selection

Fix incorrect register update that results in unintentionally
in default eSPI frequency (20MHz) whenver IO mode is changed.

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

dcpleung commented on samples/drivers/espi/src/main.c:
```diff
@@ -33,6 +33,10 @@ LOG_MODULE_DECLARE(espi, CONFIG_ESPI_LOG_LEVEL);
 
 /* 20 MHz */
 #define MIN_ESPI_FREQ         20u
+/* 25 MHz */
+#define NON_DEFAULT_FREQ      25u
+/* 66 MHz */
+#define MAX_ESPI_FREQ         66u
```
I think changing these `*_FREQ` macros to actual frequency numbers would be better, e.g. `ESPI_FREQ_20MHZ` and `ESPI_FREQ_25MHZ`. This would make it a lot clear when reading the struct below, without the need to scroll back to the top of the file to find what `MIN_ESPI_FREQ` is.

albertofloyd committed:
samples: drivers: espi: Use non-default frequency.

Use non-default frequency 25MHz and Quad mode to ensure
frequency configuration is taking effect.

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

franciscomunoz reviewed (approved):
Just change the macro like @dcpleung indicated

albertofloyd commented on samples/drivers/espi/src/main.c:
```diff
@@ -31,8 +31,12 @@ LOG_MODULE_DECLARE(espi, CONFIG_ESPI_LOG_LEVEL);
+/* Non-default eSPI frequency */
+#define ESPI_FREQ_25MHZ      25u
+/* Maximum frequency supported by eSPI bus per specification */
+#define MAX_ESPI_FREQ         66u
```
updated

albertofloyd commented on samples/drivers/espi/src/main.c:
```diff
@@ -31,8 +31,12 @@ LOG_MODULE_DECLARE(espi, CONFIG_ESPI_LOG_LEVEL);
+/* Minimum frequency supported by eSPI bus per specification*/
+#define ESPI_FREQ_20MHZ       20u
+/* Non-default eSPI frequency */
+#define ESPI_FREQ_25MHZ      25u
```
updated
